src/components/TextField/TextField.scss

```diff
@@ -60,31 +60,38 @@ $stacking-order: (
 
   // stylelint-disable-next-line selector-max-class, selector-max-combinators
   > .Input ~ .Backdrop {
-    background-color: color('red', 'lighter');
-    border-color: color('red', 'dark');
+    background-color: var(
+      --p-critical-surface-subdued,
+      color('red', 'lighter')
+    );
+    border-color: var(--p-critical-divider, color('red', 'dark'));
 
     // stylelint-disable-next-line selector-max-class, selector-max-combinators, selector-max-specificity
     &::after {
-      border-color: color('red', 'dark');
+      border-color: var(--p-interactive-focus, color('red', 'dark'));
     }
   }
 }
 
+.readOnly {
+  color: var(--p-text-disabled-on-surface);
+}
+
 .readOnly > .Backdrop {
-  background-color: color('sky', 'lighter');
+  background-color: var(--p-neutral-action-disabled, color('sky', 'lighter'));
```

Comment:
Feels like the wrong role. `--p-surface-foreground-subdued`? Or maybe another/new role?

Reply:
Or maybe not, I can see the logic here.